sklearn/linear_model/least_angle.py

```diff
@@ -1437,7 +1438,7 @@ def fit(self, X, y, copy_X=True):
 
         self.alphas_ = alphas_
         with np.errstate(divide='ignore'):
-            self.criterion_ = n_samples * np.log(mean_squared_error) + K * df
+            self.criterion_ = n_samples * mean_squared_error/sigma + K * df
```

Comment:
Thanks for your reply to #7692. In general, it is hard to estimate sigma2 with a single formula for LASSO. I looked at R implementation of [selectiveInference](https://cran.r-project.org/web/packages/selectiveInference/selectiveInference.pdf) Page 12. 

> Estimate of error standard deviation. If NULL (default), this is estimated using
the mean squared residual of the full least squares fit when n >= 2p, and using
the standard deviation of y when n < 2p.

So if n >= 2p, it should be mean_squared_error * n_samples / (n_samples - df - 1). 
if n < 2p, it should be var(y).

For the diabetes data in travis test, n >= 2p, the first formula should be used. This is probably why travis is not happy.